### PR TITLE
Change sign of the value returned by CountsStatistic.compute_errn

### PR DESCRIPTION
**Pull request by adonath**
**Description**

This pull request changes the sign return by `CountsStatistic.compute_errn` for consistency with all the other estimators in Gammapy. I adapted the description in the estimator docs as well. It also include the fix to address the recent test fails with plotting of negative error in Matplotlib 3.6.

### CODE REVIEW

**Comment by adonath** [comment]
The failed tests will be fixed in #4086, after this PR is merged. However we should get green CI for this PR now.

File: docs/user-guide/estimators.rst [hunk 1]
```diff
@@ -62,9 +62,9 @@ In addition, the following optional quantities can be computed:
 ================= =================================================
 Quantity          Definition
 ================= =================================================
-norm_errp         Positive error of the norm
-norm_errn         Negative error of the norm
-norm_ul           Upper limit of the norm
+norm_errp         Positive error of the norm, given as difference to the best fit norm
+norm_errn         Negative error of the norm, given as difference to the best fit norm
+norm_ul           Upper limit of the norm, fiven as abolsute value
```
Comment (registerrier) [inline docs/user-guide/estimators.rst:67]: given as absolute value

Note: even if negative?

**Review by registerrier — COMMENTED** [review]
Thanks @adonath .

I assume this can indeed be made consistent with the convention that errors are always positive numbers. One issue is that this is API breaking. We might consider it is worth it, but it has to be clearly visible to users.

File: docs/user-guide/estimators.rst (hunk 1 above)
Comment (adonath) [reply docs/user-guide/estimators.rst:67]: Fixed.

**Comment by adonath** [comment]
> One issue is that this is API breaking. We might consider it is worth it, but it has to be clearly visible to users.
It is not API breaking, but yes, it is backwards incompatible. I will be described in the release notes for v1.0, do you think this will be sufficient?

**Comment by adonath** [comment]
@registerrier I'll go ahead and merge this now, such that we get a green CI again. We can make additional changes to the documentation after and during preparation of the releases notes.
